Source/Core/Core/State.cpp

```diff
@@ -525,13 +506,107 @@ void SaveAs(const std::string& filename, bool wait)
       true);
 }
 
+static bool GetVersionFromLZO(StateHeader& header, File::IOFile& f)
+{
+  // Just read the first block, since it will contain the full revision string
+  lzo_uint32 cur_len = 0;  // size of compressed bytes
+  lzo_uint new_len = 0;    // size of uncompressed bytes
+  std::vector<u8> buffer;
+  buffer.resize(header.legacy_header.must_be_zero);
```

Comment:
This looks really weird... maybe rename `must_be_zero` to `lzo_size` or something like that instead, but keep a comment about it needing to be zero for newly created states.